include/upipe/ubuf_block.h

```diff
@@ -479,6 +479,50 @@ static inline int ubuf_block_resize(struct ubuf *ubuf, int offset, int new_size)
     return UBASE_ERR_NONE;
 }
 
+/** @This extends a block ubuf, if possible. This will only work if
+ * the relevant low-level buffers are not shared with other ubuf and
+ * the block manager allows to grow the buffer (ie. prepend/append have been
+ * correctly specified at allocation, or reallocation is allowed)
+ *
+ * Should this fail, @ref ubuf_block_merge may be used to achieve the same
+ * goal with an extra buffer copy.
+ *
+ * @param ubuf pointer to ubuf
+ * @param prepend number of octets to prepend
+ * @param append number of octets to append
+ * @return an error code
+ */
+static inline int ubuf_block_extend(struct ubuf *ubuf, int prepend, int append)
+{
+    assert(prepend >= 0);
+    assert(append >= 0);
+    if (ubuf->mgr->signature != UBUF_ALLOC_BLOCK)
+        return UBASE_ERR_INVALID;
+
+    if (!ubase_check(ubuf_control(ubuf, UBUF_SINGLE)))
```

Comment:
I don't understand, ubuf_block_write is not called in this function.
We wouldn't want to change an ubuf used by another uref.

That UBUF_SINGLE check was added in 5b7253903251